lib/battle/CBattleInfoEssentials.cpp

```diff
@@ -0,0 +1,359 @@
+			return true;
+	}
+
+	return false;
+}
+
+TStacks CBattleInfoEssentials::battleGetAllStacks(bool includeTurrets /*= false*/) const
+{
+	return battleGetStacksIf([=](const CStack * s)
+	{
+		return !s->isGhost() && (includeTurrets || !s->isTurret());
+	});
+}
+
+TStacks CBattleInfoEssentials::battleGetStacksIf(TStackFilter predicate) const
+{
+	TStacks ret;
+	RETURN_IF_NOT_BATTLE(ret);
+
+	vstd::copy_if(getBattle()->stacks, std::back_inserter(ret), predicate);
+
+	return ret;
+}
+
+TStacks CBattleInfoEssentials::battleAliveStacks() const
+{
+	return battleGetStacksIf([](const CStack * s){
+		return s->isValidTarget(false);
+	});
+}
+
+TStacks CBattleInfoEssentials::battleAliveStacks(ui8 side) const
+{
+	return battleGetStacksIf([=](const CStack * s){
+		return s->isValidTarget(false) && s->attackerOwned == !side;
+	});
+}
+
+int CBattleInfoEssentials::battleGetMoatDmg() const
+{
+	RETURN_IF_NOT_BATTLE(0);
+
+	auto town = getBattle()->town;
+	if(!town)
+		return 0;
+
+	return town->town->moatDamage;
+}
+
+const CGTownInstance * CBattleInfoEssentials::battleGetDefendedTown() const
+{
+	RETURN_IF_NOT_BATTLE(nullptr);
+
+
+	if(!getBattle() || getBattle()->town == nullptr)
+		return nullptr;
+
+	return getBattle()->town;
+}
+
+BattlePerspective::BattlePerspective CBattleInfoEssentials::battleGetMySide() const
+{
+	RETURN_IF_NOT_BATTLE(BattlePerspective::INVALID);
+	if(!player || player.get().isSpectator())
+		return BattlePerspective::ALL_KNOWING;
+	if(*player == getBattle()->sides[0].color)
+		return BattlePerspective::LEFT_SIDE;
+	if(*player == getBattle()->sides[1].color)
+		return BattlePerspective::RIGHT_SIDE;
+
+	logGlobal->errorStream() << "Cannot find player " << *player << " in battle!";
+	return BattlePerspective::INVALID;
+}
+
+const CStack * CBattleInfoEssentials::battleActiveStack() const
+{
+	RETURN_IF_NOT_BATTLE(nullptr);
+	return battleGetStackByID(getBattle()->activeStack);
+}
+
+const CStack* CBattleInfoEssentials::battleGetStackByID(int ID, bool onlyAlive) const
+{
+	RETURN_IF_NOT_BATTLE(nullptr);
+
+	auto stacks = battleGetStacksIf([=](const CStack * s)
+	{
+		return s->ID == ID && (!onlyAlive || s->alive());
+	});
+
+	if(stacks.empty())
+		return nullptr;
+	else
+		return stacks[0];
+}
+
+bool CBattleInfoEssentials::battleDoWeKnowAbout(ui8 side) const
+{
+	RETURN_IF_NOT_BATTLE(false);
+	auto p = battleGetMySide();
+	return p == BattlePerspective::ALL_KNOWING  ||  p == side;
+}
+
+si8 CBattleInfoEssentials::battleTacticDist() const
+{
+	RETURN_IF_NOT_BATTLE(0);
+	return getBattle()->tacticDistance;
+}
+
+si8 CBattleInfoEssentials::battleGetTacticsSide() const
+{
+	RETURN_IF_NOT_BATTLE(-1);
+	return getBattle()->tacticsSide;
+}
+
+const CGHeroInstance * CBattleInfoEssentials::battleGetFightingHero(ui8 side) const
+{
+	RETURN_IF_NOT_BATTLE(nullptr);
+	if(side > 1)
+	{
+		logGlobal->errorStream() << "FIXME: " <<  __FUNCTION__ << " wrong argument!";

```

Comment:
Rewrite all logging to new format:
`logGlobal->error("Something %s something", __FUNCTION__);`